src/test/java/bio/terra/common/iam/AuthenticatedUserRequestTest.java

```diff
@@ -0,0 +1,158 @@
+   * serialization incompatibilities at dev time.
+   * </pre>
+   */
+  private static final ObjectMapper objectMapper =
+      new ObjectMapper()
+          .registerModule(new ParameterNamesModule())
+          .registerModule(new Jdk8Module())
+          .registerModule(new JavaTimeModule())
+          .registerModule(new JsonNullableModule())
+          .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+          .enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL);
+
+  private static final String EmailAddress = "test@example.com";
+  private static final UUID RequestId = UUID.randomUUID();
+  private static final String SubjectId = "Subject";
+  private static final String Token = "0123.456-789AbCd";
+
+  @Test
+  public void fullBuilder() throws Exception {
+    AuthenticatedUserRequest req =
+        AuthenticatedUserRequest.builder()
+            .setEmail(EmailAddress)
+            .setReqId(RequestId)
+            .setSubjectId(SubjectId)
+            .setToken(Token)
+            .build();
+
+    assertEquals(EmailAddress, req.getEmail());
+    assertEquals(SubjectId, req.getSubjectId());
+    assertEquals(Token, req.getToken());
+    assertEquals(Token, req.getRequiredToken());
+    assertNotNull(req.getReqId());
+    validateJsonSerialization(req);
+  }
+
+  @Test
+  public void emptyBuilder() throws Exception {
+    AuthenticatedUserRequest req = AuthenticatedUserRequest.builder().build();
+    assertNull(req.getEmail());
+    assertNull(req.getSubjectId());
+    assertNull(req.getToken());
+    assertNotNull(req.getReqId());
+    assertThrows(ApiException.class, req::getRequiredToken);
+    validateJsonSerialization(req);
+  }
+
+  @Test
+  public void equality() {
+    AuthenticatedUserRequest req =
+        AuthenticatedUserRequest.builder()
+            .setEmail(EmailAddress)
+            .setReqId(RequestId)
+            .setSubjectId(SubjectId)
+            .setToken(Token)
+            .build();
+
+    // Positive test
+    AuthenticatedUserRequest cmp = req.toBuilder().build();
+    assertEquals(req, cmp);
+    assertEquals(req.hashCode(), cmp.hashCode());
+
+    // Negative tests
+    assertNotEquals(req, req.toBuilder().setEmail("JUNK").build());
+    assertNotEquals(req, req.toBuilder().setReqId(UUID.randomUUID()).build());
+    assertNotEquals(req, req.toBuilder().setSubjectId("JUNK").build());
+    assertNotEquals(req, req.toBuilder().setToken("JUNK").build());
+
+    // Explicit test for comparison to self
+    assertEquals(req, req);
+
+    // Explicit test for off-type comparison
+    assertNotEquals(req, "test");
+  }
+
+  static class TestVector {
+    private final String json;
+    private final AuthenticatedUserRequest request;
+
+    public TestVector(String json, AuthenticatedUserRequest request) {
+      this.json = json;
+      this.request = request;
+    }
+
+    public void validate() throws JsonProcessingException {
```

Comment:
optional style: I'm not totally convinced this is worth having a class for. I think making the `validate` function into `assertDeserializedEquals(String json, AuthenticatedUserRequest expectedRequest)` (or some better function name) and inlining the `testVectors` would keep the indirection in tests to a minimum. But this is very much on the level of stylistic opinions, and there might be a well established pattern here I'm not familiar with :)

Reply:
Yeah, in my mind I was thinking this would look a lot more compact than it turned out; also the list of test vectors will not be cumulative so there's not much to be gained by adding syntactic sugar even if it had panned out.  Simplified it.